src/algebra/group_power/order.lean

```diff
@@ -356,6 +356,22 @@ end
 theorem abs_le_of_sq_le_sq' (h : x^2 ≤ y^2) (hy : 0 ≤ y) : -y ≤ x ∧ x ≤ y :=
 abs_le.mp $ abs_le_of_sq_le_sq h hy
 
+lemma sq_eq_sq_iff_abs_eq_abs (x y : R) : x^2 = y^2 ↔ |x| = |y| :=
+⟨λ h, (abs_le_abs_of_sq_le_sq h.le).antisymm (abs_le_abs_of_sq_le_sq h.ge),
+ λ h, by rw [←sq_abs, h, sq_abs]⟩
+
+@[simp] lemma sq_eq_one_iff (x : R) : x^2 = 1 ↔ x = 1 ∨ x = -1 :=
+by rw [←abs_eq_abs, ←sq_eq_sq_iff_abs_eq_abs, one_pow]
+
+lemma sq_ne_one_iff (x : R) : x^2 ≠ 1 ↔ x ≠ 1 ∧ x ≠ -1 :=
+(not_iff_not.2 (sq_eq_one_iff _)).trans not_or_distrib
+
+@[simp] lemma sq_le_one_iff_abs_le (x : R) : x^2 ≤ 1 ↔ |x| ≤ 1 :=
+have t : x^2 ≤ 1^2 ↔ |x| ≤ |1| := ⟨abs_le_abs_of_sq_le_sq, sq_le_sq⟩, by simpa using t
+
+@[simp] lemma sq_lt_one_iff_abs_lt (x : R) : x^2 < 1 ↔ |x| < 1 :=
```

Comment:
should we also bother with a version for `≥` and `>`?

Reply:
just to be clear I meant as in `1 < x^2` not the forbidden operators

Reply:
Ah yeah I realised this too late! I think that's worth doing, I'll add it.

Reply:
Done, thanks!